test/e2e/operator.go

```diff
@@ -302,27 +303,51 @@ var _ = Describe("ARO Operator - Azure Subnet Reconciler", func() {
 	var testnsg mgmtnetwork.SecurityGroup
 
 	const nsg = "e2e-nsg"
+	const emptyMachineSet = "e2e-test-machineset"
 
 	gatherNetworkInfo := func(ctx context.Context) {
 		By("gathering vnet name, resource group, location, and adds master/worker subnets to list to reconcile")
 		oc, err := clients.OpenshiftClusters.Get(ctx, vnetResourceGroup, clusterName)
 		Expect(err).NotTo(HaveOccurred())
 		location = *oc.Location
 
-		vnet, masterSubnet, err := apisubnet.Split(*oc.OpenShiftClusterProperties.MasterProfile.SubnetID)
-		Expect(err).NotTo(HaveOccurred())
-		_, workerSubnet, err := apisubnet.Split((*(*oc.OpenShiftClusterProperties.WorkerProfiles)[0].SubnetID))
+		vnet, workerSubnet, err := apisubnet.Split((*(*oc.OpenShiftClusterProperties.WorkerProfiles)[0].SubnetID))
 		Expect(err).NotTo(HaveOccurred())
 
 		subnetsToReconcile = map[string]*string{
-			masterSubnet: to.StringPtr(""),
```

Comment:
What's the justification from removing the master subnet from e2e scope? It seems like we should be testing that since it's still part of the controller's functionality, even though we've changed it a bit.

Reply:
The primary goal was flake defense - I was noticing that if these e2e tests failed for any reason, they would leave the fake/garbage NSG on the subnets, and would effectively hose the cluster. I think, with the additional guards in place on this test, that it may be safe to reintroduce the master subnet to this test. I can try that out and hammer the test to make sure it doesn't bring back any flakes or otherwise degrade the test cluster.

Reply:
Re-introduced master subnet, e2e test passed locally w/o flakes (ran 50 times successfully in a row).